src/Features/CSharp/Portable/SignatureHelp/AbstractCSharpSignatureHelpProvider.cs

```diff
@@ -72,5 +100,308 @@ protected IList<TaggedText> GetAwaitableUsage(IMethodSymbol method, SemanticMode
+                    {
+                        inParams = true;
+                    }
+
+                    argToParamMap[argumentIndex] = parameterIndex;
+                }
+            }
+
+            bool IsEmptyArgument(ExpressionSyntax expression)
+            {
+                return expression.IsKind(SyntaxKind.IdentifierName) &&
+                    ((IdentifierNameSyntax)expression).Identifier.ValueText.Length == 0;
+            }
+
+            // If the cursor is pointing at an argument for which we did not find the corresponding
+            // parameter, we will highlight the first unspecified parameter.
+            int FirstUnspecifiedParameter()
+            {
+                var specified = ArrayBuilder<bool>.GetInstance(argumentCount, false);
+                for (var i = 0; i < argumentCount; i++)
+                {
+                    var parameterIndex = argToParamMap[i];
+                    if (parameterIndex >= 0)
+                    {
+                        specified[parameterIndex] = true;
+                    }
+                }
+
+                var first = specified.FindIndex(s => !s);
+                specified.Free();
+                return first;
+            }
+
+            // Determines if the given argument is compatible with the given parameter
+            bool IsCompatibleArgument(ArgumentSyntax argument, IParameterSymbol parameter)
+            {
+                var parameterRefKind = parameter.RefKind;
+                if (parameterRefKind == RefKind.None)
+                {
+                    if (IsEmptyArgument(argument.Expression))
+                    {
+                        return true;
+                    }
+
+                    var type = parameter.Type;
+                    if (parameter.IsParams && semanticFactsService.CanConvert(semanticModel, argument.Expression, ((IArrayTypeSymbol)type).ElementType))
+                    {
+                        return true;
+                    }
+
+                    return semanticFactsService.CanConvert(semanticModel, argument.Expression, type);
+                }
+
+                // We don't have an API to check conversion between type symbols, so we just check ref kind
+                var argumentRefKind = argument.GetRefKind();
+                if (parameterRefKind == RefKind.In && argumentRefKind == RefKind.None)
+                {
+                    return true;
+                }
+
+                if (parameterRefKind == argumentRefKind)
+                {
+                    return true;
+                }
+
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Given the cursor position, find which argument is active.
+        /// This will be useful to later find which parameter should be highlighted.
+        /// </summary>
+        protected static int GetArgumentIndex(SeparatedSyntaxList<ArgumentSyntax> arguments, int position)
```

Comment:
i'm surprised VB doesn't need the same helper.  if it does, thsi could be an extension method on `SeparateSyntaxList<T>` in the IDE layer.

Reply:
Haven't implemented the feature for VB yet. We should discuss